### PR TITLE
Cdk pipeline

### PR DESCRIPTION
**Pull request by alinadima**

### CODE REVIEW

File: demos-pipeline/lib/lib/demo-pipeline.ts [hunk 1]
```diff
@@ -146,6 +153,56 @@ export class DemoPipelineStack extends cdk.Stack {
...
+import boto3
+
+codecommit_client = boto3.client('codecommit')
+ecr_client = boto3.client('ecr')
+codepipeline_client = boto3.client('codepipeline')
+
+def handler(event, context):
+    response = ecr_client.describe_images(repositoryName='${props.imageRepo.repositoryName}', filter={'tagStatus': 'TAGGED'})
+    for i in response['imageDetails']:
+      if '${props.imageTag}' in i['imageTags']:
```
Comment (nateglims) [inline demos-pipeline/lib/lib/demo-pipeline.ts:172]: This goes back to two space tabs

File: demos-pipeline/lib/lib/demo-pipeline.ts [hunk 2]
```diff
@@ -146,6 +153,56 @@ export class DemoPipelineStack extends cdk.Stack {
...
+      resources: [props.imageRepo.repositoryArn],
+    });
+
+    fn.role?.attachInlinePolicy(
+        new iam.Policy(this, 'CheckOSAndStart', {
+          statements: [startPipelinePolicy, ecrPolicy],
+        }),
+    );
+
+    const ruleOnCreateOrUpdate = new events.Rule(this, 'BuildTriggerRule', {
```
Comment (nateglims) [inline demos-pipeline/lib/lib/demo-pipeline.ts:194]: can you run `npm run format`? Our CI check is broken...

Comment (alinadima) [reply demos-pipeline/lib/lib/demo-pipeline.ts:194]: I think we should compare and align the local formatting settings. I might have some old stuff configured in Intelij for js and ts. I will take a look.

Comment (nateglims) [reply demos-pipeline/lib/lib/demo-pipeline.ts:194]: With the plugins, eslint/prettier should format correctly. VS Code will handle it with the prettier plugin, but JS formatting is excessively complicated.

Comment (alinadima) [reply demos-pipeline/lib/lib/demo-pipeline.ts]: This should be good now. We should add this automatically in npm run build perhaps.

**Review by nateglims — APPROVED** [review]

**Pull request merged**